OptimizelySDKCore/OptimizelySDKCore/OPTLYNotificationCenter.h

```diff
@@ -26,20 +26,28 @@ typedef NS_ENUM(NSUInteger, OPTLYNotificationType) {
 
 typedef void (^ActivateListener)(OPTLYExperiment * _Nonnull experiment,
                                  NSString * _Nonnull userId,
-                                 NSDictionary<NSString *, NSObject *> * _Nonnull attributes,
+                                 NSDictionary<NSString *, NSObject *> * _Nullable attributes,
                                  OPTLYVariation * _Nonnull variation,
                                  NSDictionary<NSString *,NSObject *> * _Nonnull event);
 
 typedef void (^TrackListener)(NSString * _Nonnull eventKey,
                               NSString * _Nonnull userId,
-                              NSDictionary<NSString *, NSObject *> * _Nonnull attributes,
-                              NSDictionary * _Nonnull eventTags,
+                              NSDictionary<NSString *, NSObject *> * _Nullable attributes,
+                              NSDictionary * _Nullable eventTags,
                               NSDictionary<NSString *,NSObject *> * _Nonnull event);
 
-typedef void (^GenericListener)(NSArray * _Nonnull args);
+typedef void (^GenericListener)(NSDictionary * _Nonnull args);
 
 typedef NSMutableDictionary<NSNumber *, GenericListener > OPTLYNotificationHolder;
 
+static NSString *const OPTLYNotificationExperimentKey = @"experiment";
+static NSString *const OPTLYNotificationVariationKey = @"variation";
+static NSString *const OPTLYNotificationUserIdKey = @"userId";
+static NSString *const OPTLYNotificationAttributesKey = @"attributes";
+static NSString *const OPTLYNotificationEventKey = @"eventKey";
+static NSString *const OPTLYNotificationEventTags = @"eventTags";
```

Comment:
Not sure about naming rules, but to be consistent, they should like:
OPTLYNotificationEventTags --> OPTLYNotificationEventTagsKey
OPTLYNotificationLogEventParams --> OPTLYNotificationLogEventParamsKey